src/neo/IO/Caching/DataCache.cs

```diff
@@ -139,7 +149,10 @@ public void DeleteWhere(Func<TKey, TValue, bool> predicate)
             lock (dictionary)
             {
                 foreach (Trackable trackable in dictionary.Where(p => p.Value.State != TrackState.Deleted && predicate(p.Key, p.Value.Item)).Select(p => p.Value))
+                {
+                    changeSet.Add(trackable.Key);
                     trackable.State = TrackState.Deleted;
+                }
```

Comment:
Is it a bug, we set state `TrackState.Deleted` directly without check where it is `TrackState.Added`? @erikzhang

Reply:
Yes, I think so. And is this method used in somewhere?

Reply:
No one use it. Let's remove it?

Reply:
if we don't need it , it could be removed

Reply:
Removed, some ut have also been adjusted. BTW, if we want to keep this method, we should rename it, because it's only delete data in the cache `dictionary`, which is different from `Delete` method.